File: ClipboardExpense/ClipboardExpense/Models/ExpenseRecord.swift
import Foundation
import SwiftData

@Model
final class ExpenseRecord {
    var id: UUID
    var amount: Decimal
    var category: String
    var note: String?
    var merchant: String?
    var time: Date
    var source: String
    var rawText: String?

    init(
        id: UUID = UUID(),
        amount: Decimal,
        category: String,
        note: String? = nil,
        merchant: String? = nil,
        time: Date = .now,
        source: String = "manual",
        rawText: String? = nil
    ) {
        self.id = id
        self.amount = amount
        self.category = category
        self.note = note
        self.merchant = merchant
        self.time = time
        self.source = source
        self.rawText = rawText
    }
}

File: ClipboardExpenseAndroid/app/src/main/java/com/shuimingbai/clipboardexpense/data/ExpenseRecord.kt
package com.shuimingbai.clipboardexpense.data

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.math.BigDecimal
import java.util.Date

@Entity(tableName = "expense_records")
data class ExpenseRecord(
    @PrimaryKey val id: String = java.util.UUID.randomUUID().toString(),
    val amount: BigDecimal,
    val category: String,
    val note: String? = null,
    val merchant: String? = null,
    val time: Date,
    val source: String,
    val rawText: String? = null
)
